README.md

```diff
@@ -234,6 +235,40 @@ ROS 2 allows controlling participant discovery with two environment variables: `
 Full configuration of participant discovery can also be set with XML files; however, the ROS specific environment variables should be disabled to prevent them from interfering.
 Set `ROS_AUTOMATIC_DISCOVERY_RANGE` to the value `SYSTEM_DEFAULT` to disable both ROS specific environment variables.
 
+### Enable Zero Copy Data Sharing
+
+ROS 2 provides [Loaned Messages](https://design.ros2.org/articles/zero_copy.html) that allows the user application to loan the message memory from the RMW implementation to eliminate the copy between the ROS 2 application and RMW implementation.
+And Fast DDS `rmw_fastrtps_cpp` provides [Shared Memory Transport](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/shared_memory/shared_memory.html) and [Data-sharing delivery](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/datasharing.html) features to speed up the localhost communication.
+Taking advantage of these features all together, it provides significant performance improvement to ROS 2 application.
+
+By default, `rmw_fastrtps_cpp` tries to use [Shared Memory Transport](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/shared_memory/shared_memory.html) and [Data-sharing delivery](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/datasharing.html) for localhost communication along with network communication if the message data type is a bounded type (a fixed sized data object).
```

Comment:
It seems that I'm late, but this is not entirely true. By default `rmw_fastrtps` sets data-sharing to `OFF` [here](https://github.com/ros2/rmw_fastrtps/blob/rolling/rmw_fastrtps_cpp/src/publisher.cpp#L255) and [here](https://github.com/ros2/rmw_fastrtps/blob/rolling/rmw_fastrtps_dynamic_cpp/src/publisher.cpp#L261). To enable it, users need to:

1. Set the XML
2. `export RMW_FASTRTPS_USE_QOS_FROM_XML=1`.

I'll make a PR to adjust the text

Reply:
@EduPonz really? i did the test to see the files under `/dev/shm` with current rolling. you mean the data did not go through the shared memory mapped files?

Reply:
So i did some experimental tests on local environment. i think i was looking at the files for `shared memory transport` but `data sharing`. as you mentioned, if i enable `data sharing` via xml, i can see the file like `fast_datasharing_01.0f.60.db.97.c9.17.1a.00.00.00.00_0.0.14.3`.

thanks for catching up on this, really helped!

Reply:
Yeah, it is a rather common misunderstanding, and at honest one as that, as the terms can very easily get mixed up. In short, Fast DDS has two shared memory based delivery mechanism:

1. One is what we call a full compliant transport, which is similar to a socket based transport in the sense that you have ports and encapsulation of the data that is copied from the data writers local history cache into the transport buffers. This can be used without any kind of type restrictions. Out of the box, Fast DDS participants use one instance of this transport to communicate with other participants that are detected to be on the same machine.
1. Then we have what we called data sharing delivery mechanism. It is still based on memory mapped files (same as the SHM), but in here, under some conditions, the data writer stores its history cache in the mmap file directly, and the data readers read the data directly from there, thus removing all the transport related copies. This approach however comes with data type limitations, mainly that data is required to be bounded, as the nmap segment needs to be able to store all the writer's history cache. Mind that this is not the case with SHM transport, as the serialized data can be fragmented if the payload is larger than the SHM buffer. If both a data writer and reader in the same host are data sharing compatible, meaning that their data types are bounded and they have the qos set to `on` or `auto`, then they will communicate over data sharing instead of the SHM transport.

Because of the previous, Zero Copy can only be achieved by combining sample loaning with data sharing, but not with SHM transport.